lib/oat/serializer.rb

```diff
@@ -65,5 +77,16 @@ def map_property(name)
       property name, value
     end
 
+    def set_serialized(type, id)
+      @context[:_serialized_entities][type] ||= {}
+      @context[:_serialized_entities][type][id] = true
+    end
+
+    def serialized?(type, id)
+      h = @context[:_serialized_entities]
+      h = h[type] if h
+      h[id] if h
```

Comment:
This method should always return `true` or `false`, so maybe the last line could be:

```ruby
!!h[id]
```